src/remote.ts

```diff
@@ -34,13 +34,28 @@ export class NoSSHGatewayError extends Error {
 
 export const SSH_DEST_KEY = 'ssh-dest:';
 
-export function getGitpodRemoteWindowConnectionInfo(context: vscode.ExtensionContext): { remoteAuthority: string; connectionInfo: SSHConnectionParams } | undefined {
+type SSHConnType = 'local-app' | 'local-ssh' | 'ssh-gateway';
+export interface RemoteConnectionInfo {
+    remoteAuthority: string;
+    connectionInfo: SSHConnectionParams;
+    connType: SSHConnType;
+}
+
+export function getGitpodRemoteWindowConnectionInfo(context: vscode.ExtensionContext): RemoteConnectionInfo | undefined {
     const remoteUri = vscode.workspace.workspaceFile || vscode.workspace.workspaceFolders?.[0].uri;
     if (vscode.env.remoteName === 'ssh-remote' && context.extension.extensionKind === vscode.ExtensionKind.UI && remoteUri) {
         const [, sshDestStr] = remoteUri.authority.split('+');
+        const sshDestInfo = JSON.parse(Buffer.from(sshDestStr, 'hex').toString()) as { hostName: string; user?: string };
         const connectionInfo = context.globalState.get<SSHConnectionParams>(`${SSH_DEST_KEY}${sshDestStr}`);
         if (connectionInfo) {
-            return { remoteAuthority: remoteUri.authority, connectionInfo };
+            let connType: SSHConnType = 'local-app';
+            const domain = getLocalSSHDomain(connectionInfo.gitpodHost);
+            if (sshDestInfo.hostName.endsWith('.' + domain)) {
+                connType = 'local-ssh';
+            } else if (!sshDestInfo.hostName.includes('.')) {
```

Comment:
shouldn't we check for `.ssh.` instead? '.' feels trivial

Reply:
if with local-app, we don't have any '.' but with `.ssh`, it may have a Dedicated host with `ssh`? @jeanp413 @akosyakov

Reply:
local-app has port right? you can get it from `sshDestInfo` then

Reply:
no idea :thinking: leave it up to you then, but let's rename `local-ssh` to something else is no longer local-ssh maybe `ssh-tunnel` or `ssh-proxy`

Reply:
Resolved by identify it in `RemoteConnector`